### PR TITLE
draft version of bn265_algebra

### PR DESCRIPTION
**Pull request by nanne007**
Currently,all the natives(except hash_to_curve) now support bn254 structures.
docs and tests are still missing.
and the gas of bn254 algebra operations are wrong.
I have tried the doc in `scripts/algebra-gas/README.md`, but the results of bls12_381 are different from the existing ones.
so I think there may be some gaps here.

I draft the pr here to get comments from outside, so that I can continue to improve the code to make the pr acceptable by aptos.

### CODE REVIEW

File: aptos-move/framework/src/natives/cryptography/algebra/serialization.rs [hunk 1]
```diff
@@ -180,6 +185,161 @@ pub fn serialize_internal(
...
+        },
+        (BN254Structure::BN254Fq12, BN254SerializationFormat::BN254Fq12LscLsb) => {
+            let handle = safely_pop_arg!(args, u64) as usize;
+            safe_borrow_element!(context, handle, ark_bn254::Fq12, element_ptr, element);
+            let mut buf = vec![];
+            context.charge(ALGEBRA_ARK_BN254_FQ12_SERIALIZE)?;
+            element
+                .serialize_uncompressed(&mut buf)
+                .map_err(|_e| abort_invariant_violated())?;
+            buf.reverse();
```
Comment (zjma) [inline aptos-move/framework/src/natives/cryptography/algebra/serialization.rs]: unlike BSL12381Fq12LscLsb, here a reverse is needed? Let me double check.

**Comment by zjma** [comment]
Awesome work! Took a super quick look, no major issue.

I can help the gas calibration later in the official PR.

**Comment by zjma** [comment]
Can you try open an official PR against aptos-core/main branch? there we can run more tests.

File: aptos-move/framework/src/natives/cryptography/algebra/serialization.rs (hunk 1 above)
Comment (nanne007) [reply aptos-move/framework/src/natives/cryptography/algebra/serialization.rs]: Oh, nice catch! that's a bug! I may wrongly copy pasted from msb part.

Comment (nanne007) [reply aptos-move/framework/src/natives/cryptography/algebra/serialization.rs]: fixed now.

**Pull request closed without merging**